### PR TITLE
[parsec-cli] Read some options from env variables

### PR DESCRIPTION
**Pull request by FirelightFlagboy**
- Add `env` feature to clap dependency
- Use `env` feature to read some options from env variables:

  - `PARSEC_SERVER_ADDR` for server address
  - `PARSEC_ADMINISTRATION_TOKEN` for administration token
  - `PARSEC_CONFIG_DIR` for Parsec config directory
  - `PARSEC_DEVICE_ID` for device ID

### CODE REVIEW

File: cli/src/cancel_invitation.rs [hunk 1]
```diff
@@ -1,34 +1,31 @@
...
 
 pub async fn cancel_invitation(cancel_invitation: CancelInvitation) -> anyhow::Result<()> {
     let CancelInvitation {
-        config_dir,
-        device,
+        config: ConfigWithDeviceSharedOpts { config_dir, device },
         token,
     } = cancel_invitation;
+    log::trace!(
+        "Cancelling invitation (confdir={}, device={})",
```
Comment (mmmarcos) [inline cli/src/cancel_invitation.rs:25]:
```suggestion
        "Cancelling invitation (config_dir={}, device={})",
```

File: cli/src/export_recovery_device.rs [hunk 2]
```diff
@@ -24,10 +19,15 @@ pub async fn export_recovery_device(
     export_recovery_device: ExportRecoveryDevice,
 ) -> anyhow::Result<()> {
     let ExportRecoveryDevice {
-        config_dir,
-        device,
+        config: ConfigWithDeviceSharedOpts { config_dir, device },
         output,
     } = export_recovery_device;
+    log::trace!(
+        "Exporting recovery device at {} (confdir={}, device={})",
```
Comment (mmmarcos) [inline cli/src/export_recovery_device.rs:26]:
```suggestion
        "Exporting recovery device at {} (config_dir={}, device={})",
```

File: cli/src/greet_invitation.rs [hunk 3]
```diff
@@ -17,25 +15,25 @@ use libparsec::{
...
 
 pub async fn greet_invitation(greet_invitation: GreetInvitation) -> anyhow::Result<()> {
     let GreetInvitation {
-        config_dir,
-        device,
+        config: ConfigWithDeviceSharedOpts { config_dir, device },
         token,
     } = greet_invitation;
+    log::trace!(
+        "Greeting invitation (confdir={}, device={})",
```
Comment (mmmarcos) [inline cli/src/greet_invitation.rs:33]:
```suggestion
        "Greeting invitation (config_dir={}, device={})",
```

File: cli/src/import_recovery_device.rs [hunk 4]
```diff
@@ -24,10 +22,15 @@ pub async fn import_recovery_device(
     import_recovery_device: ImportRecoveryDevice,
 ) -> anyhow::Result<()> {
     let ImportRecoveryDevice {
+        config: ConfigSharedOpts { config_dir },
         input,
         passphrase,
-        config_dir,
     } = import_recovery_device;
+    log::trace!(
+        "Importing recovery device from {} (confdir={})",
```
Comment (mmmarcos) [inline cli/src/import_recovery_device.rs:30]:
```suggestion
        "Importing recovery device from {} (config_dir={})",
```

File: cli/src/invite_device.rs [hunk 5]
```diff
@@ -1,27 +1,27 @@
...
+    config: ConfigWithDeviceSharedOpts,
 }
 
 pub async fn invite_device(invite_device: InviteDevice) -> anyhow::Result<()> {
-    let InviteDevice { config_dir, device } = invite_device;
+    let InviteDevice {
+        config: ConfigWithDeviceSharedOpts { config_dir, device },
+    } = invite_device;
+    log::trace!(
+        "Inviting a device (confdir={}, device={})",
```
Comment (mmmarcos) [inline cli/src/invite_device.rs:21]:
```suggestion
        "Inviting device (config_dir={}, device={})",
```

File: cli/src/create_workspace.rs [hunk 6]
```diff
@@ -1,31 +1,28 @@
...
 
 pub async fn create_workspace(create_workspace: CreateWorkspace) -> anyhow::Result<()> {
     let CreateWorkspace {
-        config_dir,
-        device,
+        config: ConfigWithDeviceSharedOpts { config_dir, device },
         name,
     } = create_workspace;
+    log::trace!(
+        "Creating workspace {name} (confdir={}, device={})",
```
Comment (mmmarcos) [inline cli/src/create_workspace.rs:22]:
```suggestion
        "Creating workspace {name} (config_dir={}, device={})",
```

File: cli/src/list_devices.rs [hunk 7]
```diff
@@ -1,23 +1,20 @@
...
-    config_dir: PathBuf,
+    #[clap(flatten)]
+    config: ConfigSharedOpts,
 }
 
 pub async fn list_devices(list_devices: ListDevices) -> anyhow::Result<()> {
-    let config_dir = list_devices.config_dir;
-    let config_dir_str = config_dir.to_string_lossy();
+    let config_dir = list_devices.config.config_dir;
+    log::trace!("Listing devices under {}", config_dir.display());
```
Comment (mmmarcos) [inline cli/src/list_devices.rs:15]: why not use the same syntax that other messages:
```suggestion
    log::trace!("Listing devices (config_dir={})", config_dir.display());
```

File: cli/src/list_invitations.rs [hunk 8]
```diff
@@ -1,27 +1,27 @@
...
+    config: ConfigWithDeviceSharedOpts,
 }
 
 pub async fn list_invitations(list_invitations: ListInvitations) -> anyhow::Result<()> {
-    let ListInvitations { config_dir, device } = list_invitations;
+    let ListInvitations {
+        config: ConfigWithDeviceSharedOpts { config_dir, device },
+    } = list_invitations;
+    log::trace!(
+        "Listing invitations (confdir={}, device={})",
```
Comment (mmmarcos) [inline cli/src/list_invitations.rs:21]:
```suggestion
        "Listing invitations (config_dir={}, device={})",
```

File: cli/src/list_users.rs [hunk 9]
```diff
@@ -1,31 +1,26 @@
...
 
 pub async fn list_users(list_users: ListUsers) -> anyhow::Result<()> {
     let ListUsers {
-        config_dir,
-        device,
+        config: ConfigWithDeviceSharedOpts { config_dir, device },
         skip_revoked,
     } = list_users;
+    log::trace!(
+        "Listing users (confdir={}, device={}, skip_revoked={skip_revoked})",
```
Comment (mmmarcos) [inline cli/src/list_users.rs:20]:
```suggestion
        "Listing users (config_dir={}, device={}, skip_revoked={skip_revoked})",
```

File: cli/src/list_workspaces.rs [hunk 10]
```diff
@@ -1,24 +1,22 @@
...
+    config: ConfigWithDeviceSharedOpts,
 }
 
 pub async fn list_workspaces(list_workspaces: ListWorkspaces) -> anyhow::Result<()> {
-    let ListWorkspaces { config_dir, device } = list_workspaces;
+    let ListWorkspaces {
+        config: ConfigWithDeviceSharedOpts { config_dir, device },
+    } = list_workspaces;
+    log::trace!(
+        "Listing workspaces (confdir={}, device={})",
```
Comment (mmmarcos) [inline cli/src/list_workspaces.rs:16]:
```suggestion
        "Listing workspaces (config_dir={}, device={})",
```

File: cli/src/remove_device.rs [hunk 11]
```diff
@@ -1,24 +1,22 @@
...
+    config: ConfigWithDeviceSharedOpts,
 }
 
 pub async fn remove_device(remove_device: RemoveDevice) -> anyhow::Result<()> {
-    let RemoveDevice { config_dir, device } = remove_device;
+    let RemoveDevice {
+        config: ConfigWithDeviceSharedOpts { config_dir, device },
+    } = remove_device;
+    log::trace!(
+        "Removing device {device} (confdir={})",
```
Comment (mmmarcos) [inline cli/src/remove_device.rs:16]:
```suggestion
        "Removing device {device} (config_dir={})",
```

File: cli/src/share_workspace.rs [hunk 12]
```diff
@@ -28,12 +21,16 @@ pub struct ShareWorkspace {
...
     let ShareWorkspace {
-        config_dir,
-        device,
+        config: ConfigWithDeviceSharedOpts { config_dir, device },
         workspace_id,
         user_id,
         role,
     } = share_workspace;
+    log::trace!(
+        "Sharing workspace {workspace_id} to {user_id} with role {role} (confdir={}, device={})",
```
Comment (mmmarcos) [inline cli/src/share_workspace.rs:30]:
```suggestion
        "Sharing workspace {workspace_id} to {user_id} with role {role} (config_dir={}, device={})",
```

File: cli/src/stats_server.rs [hunk 13]
```diff
@@ -73,11 +70,11 @@ pub async fn stats_server_req(
 
 pub async fn stats_server(stats_organization: StatsServer) -> anyhow::Result<()> {
     let StatsServer {
-        addr,
-        token,
+        server: ServerSharedOpts { addr, token },
         format,
         end_date,
     } = stats_organization;
+    log::trace!("Retrieving server's stats (addr={addr}, format={format})");
```
Comment (mmmarcos) [inline cli/src/stats_server.rs:77]:
```suggestion
    log::trace!("Retrieving server stats (addr={addr}, format={format})");
```

File: cli/src/status_organization.rs [hunk 14]
```diff
@@ -40,9 +37,9 @@ pub async fn status_organization_req(
 pub async fn status_organization(status_organization: StatusOrganization) -> anyhow::Result<()> {
     let StatusOrganization {
         organization_id,
-        addr,
-        token,
+        server: ServerSharedOpts { addr, token },
     } = status_organization;
+    log::trace!("Retrieving status of organization {organization_id} (addr={addr})");
```
Comment (mmmarcos) [inline cli/src/status_organization.rs:42]:
```suggestion
    log::trace!("Retrieving organization status for {organization_id} (addr={addr})");
```

File: cli/src/stats_organization.rs [hunk 15]
```diff
@@ -40,9 +37,9 @@ pub async fn stats_organization_req(
 pub async fn stats_organization(stats_organization: StatsOrganization) -> anyhow::Result<()> {
     let StatsOrganization {
         organization_id,
-        addr,
-        token,
+        server: ServerSharedOpts { addr, token },
     } = stats_organization;
+    log::trace!("Retrieving stats for organization {organization_id} (addr={addr})");
```
Comment (mmmarcos) [inline cli/src/stats_organization.rs:42]:
```suggestion
    log::trace!("Retrieving organization stats for {organization_id} (addr={addr})");
```

File: cli/src/create_organization.rs [hunk 16]
```diff
@@ -74,9 +69,9 @@ pub async fn create_organization_req(
 pub async fn create_organization(create_organization: CreateOrganization) -> anyhow::Result<()> {
     let CreateOrganization {
         organization_id,
-        addr,
-        token,
+        server: ServerSharedOpts { addr, token },
     } = create_organization;
+    log::trace!("Creating organization \"{organization_id}\" (addr={addr})");
```
Comment (mmmarcos) [inline cli/src/create_organization.rs:74]: To remain consistent with other messages below:
```suggestion
    log::trace!("Creating organization {organization_id} (addr={addr})");
```

File: libparsec/crates/platform_device_loader/src/lib.rs [hunk 17]
```diff
@@ -26,9 +26,9 @@ pub const ARGON2ID_DEFAULT_PARALLELISM: u32 = 1;
 
 pub(crate) const DEVICE_FILE_EXT: &str = "keys";
 
-pub const PARSEC_CONFIG_DIR: &str = "PARSEC_CONFIG_DIR";
-pub const PARSEC_DATA_DIR: &str = "PARSEC_DATA_DIR";
-pub const PARSEC_HOME_DIR: &str = "PARSEC_HOME_DIR";
+pub const PARSEC_BASE_CONFIG_DIR: &str = "PARSEC_BASE_CONFIG_DIR";
+pub const PARSEC_BASE_DATA_DIR: &str = "PARSEC_BASE_DATA_DIR";
+pub const PARSEC_BASE_HOME_DIR: &str = "PARSEC_BASE_HOME_DIR";
```
Comment (mmmarcos) [inline libparsec/crates/platform_device_loader/src/lib.rs:31]: What is the rationale for this change? (it is not described in the PR description)

Comment (FirelightFlagboy) [reply libparsec/crates/platform_device_loader/src/lib.rs:31]: The description was put in the 3 commit, but not in the PR description

## The difference between `BASE_DIR` and `DIR`

- `BASE_DIR` as the name suggest, it's used as a base where it's folder is used to add sub-folder (`{BASE_DIR}/foo/bar`).
- `DIR`, it's used as is (`{DIR}/foo.txt`).

## Why did I made this change?

The mean reason was to be consistent, 
We had functions like `get_default_config_dir` that used `PARSEC_CONFIG_DIR` as a base value where a subpath was added to it (`{CONFIG_DIR}/parsec3/libparsec`).

That conflict with the CLI where the value will be used as is without additional transformation (i.e.: adding a subpath).

So that the reason I choose to rename those variable used in `platform-device-loader` to include `_BASE` since we add a subpath to it. 

But we could also rework the functions where the env var is used as is without adding a subpath (to copy the behavior of the CLI).

**Review by mmmarcos — APPROVED** [review]
